File: tests/integration/benchmarks/fixtures/typescript/rosetta/src/processor.ts
/**
 * Core processing logic for the Rosetta benchmark fixture.
 * 
 * This module exercises multiple import patterns:
 * - Namespace import: `import * as Models from "./models"`
 * - Selective imports: `import { format, sum } from "./helpers"`
 * - Type-only import: `import type { ProcessorConfig } from "./types"`
 */

import * as Models from "./models";
import type { ProcessorConfig, Status } from "./types";
import { format, sum, average } from "./helpers";

/** Default configuration for processing. */
const DEFAULT_CONFIG: ProcessorConfig = {
  batchSize: 100,
  timeout: 5000,
  strict: true
};

/**
 * Processes a batch of records and generates a report.
 * 
 * Uses namespace import (Models.*) to access Record and Report types,
 * demonstrating how adapters should handle wildcard imports.
 * 
 * @param records - Records to process
 * @param config - Optional processing configuration  
 * @returns Summary report of processed records
 */
export function run(
  records: Models.Record[], 
  config: ProcessorConfig = DEFAULT_CONFIG
): Models.Report {
  if (!Models.validateConfig(config)) {
    throw new Error("Invalid processor configuration");
  }

  const values = records.map(r => r.value);
  const total = sum(values);
  const avg = average(values);

  return {
    total,
    average: avg,
    records,
    generatedAt: new Date()
  };
}

/**
 * Creates a formatted summary string from a report.
 * 
 * @param report - Report to summarize
 * @returns Human-readable summary
 */
export function summarize(report: Models.Report): string {
  return `Total: ${format(report.total)}, Average: ${format(report.average)}, Count: ${report.records.length}`;
}

File: tests/integration/benchmarks/fixtures/python/rosetta/src/processor.py
"""
Core processing logic for the Rosetta benchmark fixture.

This module exercises multiple import patterns:
- Module import: `import models as Models`
- Selective imports: `from helpers import format_value, sum_values`
- Type hints: `from types import ProcessorConfig`
"""

import models as Models
from core_types import ProcessorConfig, Status
from helpers import format_value, sum_values, average
from datetime import datetime
from typing import List


# Default configuration for processing
DEFAULT_CONFIG = ProcessorConfig(
    batch_size=100,
    timeout=5000,
    strict=True
)


def run(records: List[Models.Record], 
        config: ProcessorConfig = None) -> Models.Report:
    """
    Processes a batch of records and generates a report.
    
    Uses module alias (Models) to access Record and Report types,
    demonstrating how adapters should handle aliased imports.
    
    Args:
        records: Records to process
        config: Optional processing configuration
        
    Returns:
        Summary report of processed records
    """
    if config is None:
        config = DEFAULT_CONFIG
        
    if not Models.validate_config(config):
        raise ValueError("Invalid processor configuration")

    values = [r.value for r in records]
    total = sum_values(values)
    avg = average(values)

    return Models.Report(
        total=total,
        average=avg,
        records=records,
        generated_at=datetime.now()
    )


def summarize(report: Models.Report) -> str:
    """
    Creates a formatted summary string from a report.
    
    Args:
        report: Report to summarize
        
    Returns:
        Human-readable summary
    """
    return f"Total: {format_value(report.total)}, Average: {format_value(report.average)}, Count: {len(report.records)}"
